x/distribution/keeper/keeper.go

```diff
@@ -231,6 +231,10 @@ func (k Keeper) WithdrawSingleShareRecordReward(ctx context.Context, recordID ui
 	}
 	owner := sdk.AccAddress(ownerAddr)
 
+	if k.bankKeeper.BlockedAddr(owner) {
```

Comment:
I'm a bit confused why some logic is in the msg_server and parts on the keeper? It makes more sense to me to do it all in the keeper functions.

Am I missing something here?

Reply:
i've initially put the checks right after the address is extracted. for WithdrawTokenizeShareRecordReward and WithdrawAllTokenizeShareRecordReward is in msg server (same as already existed for for CommunityPoolSpend), and here is in keeper to avoid extracting address twice but yes probably better to move it all in one place keeper or msg server. I move all in keeper as suggested (this comment is applicable to above two comments because it refers to the same thing)

Reply:
fix [47b290d](https://github.com/cosmos/cosmos-sdk/pull/22718/commits/47b290d9d88ff68e38ea202f855cd9283bcfb3d7)